### PR TITLE
Porting CoreFx changes to enable publishing to azure feeds.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* dotnet/corefx/pull/24784
* dotnet/corefx/pull/25031
* dotnet/corefx/pull/26192

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
I've made the BuildDefinition changes and am in the process of testing this in a private branch in VSTS.

@dagood @karajas Since these changes were yours initially in CoreFx would you be willing to take a quick look and let me know if you are concerned with anything as it applies to WCF?

File: dependencies.props [hunk 1]
```diff
@@ -50,6 +50,12 @@
     <PlatformPackageVersion>2.1.0-preview1-25324-02</PlatformPackageVersion>
   </PropertyGroup>
 
+  <!-- Package versions used as toolsets -->
+  <PropertyGroup>
+    <FeedTasksPackage>Microsoft.DotNet.Build.Tasks.Feed</FeedTasksPackage>
+    <FeedTasksPackageVersion>1.0.0-prerelease-02121-01</FeedTasksPackageVersion>
```
Comment (dagood) [inline dependencies.props]: This doesn't match the dotnet/versions commit you're on:

https://github.com/dotnet/versions/blob/45d075f60f80600fb1a49944239c95bc839eeab4/build-info/dotnet/buildtools/master/Last_Build_Packages.txt

Microsoft.DotNet.Build.Tasks.Feed 2.1.0-prerelease-02417-06

I suggest running `/t:UpdateDependencies` to sync things up.

File: src/publish.proj [hunk 2]
```diff
@@ -3,15 +3,16 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <Import Project="$(ToolsDir)PublishContent.targets" />
   <Import Project="$(ToolsDir)versioning.targets" />
+  <Import Project="$(PackagesDir)/$(FeedTasksPackage.ToLower())/$(FeedTasksPackageVersion)/build/$(FeedTasksPackage).targets" />
 
   <PropertyGroup>
-    <PublishPattern Condition="'$(PublishPattern)' == ''">$(PackageOutputRoot)**\*.nupkg</PublishPattern>
+    <PublishPattern Condition="'$(PublishPattern)' == ''">$(PackageOutputRoot)\**\*.nupkg</PublishPattern>
```
Comment (dagood) [inline src/publish.proj]: I was going to ask why the `\`, but I see CoreFX has it, too. Oh well, keeping in sync is best.

File: dependencies.props (hunk 1 above)
Comment (StephenBonikowsky) [reply dependencies.props]: Thanks, updated and amended.

**Comment by StephenBonikowsky** [comment]
Thanks!

**Pull request merged**